nodes/stampzilla-server/websocket/clients.go

```diff
@@ -53,14 +53,23 @@ func (r *Clients) SendToAll(t string, data interface{}) {
 
 	r.Lock()
 	defer r.Unlock()
+	clientsToRemove := make([]*Client, 0)
+
 	for _, c := range r.clients {
 		select {
 		case c.out <- msg:
 			// Everything went well :)
 		case <-time.After(time.Second):
 			log.Warn("Failed writing to websocket: timeout (", c.Name, ")")
+			clientsToRemove = append(clientsToRemove, c)
 		}
 	}
+
```

Comment:
No. I just mean unlock on the line i commented instead of defer. Thats still OK since removeClient does its own lock. You should unlock after whatever needed the lock is done. Since you add more code after the for loop the unlock needs to happen before that code.